pkg/kritis/review/validating_transport.go

```diff
@@ -101,34 +102,41 @@ func (avt *AttestorValidatingTransport) validatePublicKeyId(pubKey v1beta1.Publi
 	return nil
 }
 
-func (avt *AttestorValidatingTransport) GetValidatedAttestations(image string) ([]attestation.ValidatedAttestation, error) {
-	keys := map[string]v1beta1.PublicKey{}
-	numKeys := len(avt.Attestor.Spec.PublicKeys)
-	for i, pubKey := range avt.Attestor.Spec.PublicKeys {
-		if err := avt.validatePublicKey(pubKey); err != nil {
-			// warning level because single key failure is something tolerable
+// ParsePublicKeys fetches public keys from the Attestor spec. It returns the
+// valid PublicKeys and the keyID's for keys that could not be parsed.
+func (avt *AttestorValidatingTransport) parsePublicKeys() ([]cryptolib.PublicKey, []string) {
+	attestorKeys := avt.Attestor.Spec.PublicKeys
+	numKeys := len(attestorKeys)
+	publicKeys, invalidKeys := []cryptolib.PublicKey{}, []string{}
+
+	for i, attestorKey := range attestorKeys {
+		if attestorKey.KeyType != "PGP" {
+			glog.Warningf("Skipping key %q with unsupported type %q", attestorKey.KeyId, attestorKey.KeyType)
+			continue
+		}
+		if err := avt.validatePublicKey(attestorKey); err != nil {
+			// Warning level because single key failure is something tolerable
 			glog.Warningf("Error parsing key %d (%d keys total) for %q: %v", i, numKeys, avt.Attestor.Name, err)
+			invalidKeys = append(invalidKeys, attestorKey.KeyId)
 			continue
 		}
-		if _, ok := keys[pubKey.KeyId]; ok {
-			glog.Warningf("Overwriting key with same fingerprint %s for %q.", pubKey.KeyId, avt.Attestor.Name)
+		decodedKey, err := base64.StdEncoding.DecodeString(attestorKey.AsciiArmoredPgpPublicKey)
+		if err != nil {
+			glog.Warningf("Cannot base64 decode public key: %v", err)
+			invalidKeys = append(invalidKeys, attestorKey.KeyId)
+			continue
 		}
-		keys[pubKey.KeyId] = pubKey
-	}
-	if len(keys) == 0 {
-		return nil, fmt.Errorf("unable to find any valid key for %q", avt.Attestor.Name)
+		publicKey := cryptolib.NewPublicKey(cryptolib.Pgp, decodedKey, attestorKey.KeyId)
+		publicKeys = append(publicKeys, publicKey)
 	}
+	return publicKeys, invalidKeys
```

Comment:
NIT: would still be good to explicitly return error if len(publicKeys) == 0. How it is structured now relies on the caller function to do things properly.

Reply:
We decided offline that `parsePublicKeys` should not return an error, but should return successful and failed parsing results. We decided to provide the attestor's public keys as an argument.